src/modals/SendModals/SendModalDeployContract.tsx

```diff
@@ -195,16 +196,18 @@ const DeployContractBuildTxModalContent = ({ data, onSubmit, onCancel }: DeployC
   )
 }
 
-const handleSend = async (client: Client, txData: DeployContractTxData, context: TxContext) => {
+const handleSend = async (txData: DeployContractTxData, context: TxContext) => {
   if (!context.unsignedTransaction) throw Error('No unsignedTransaction available')
 
-  const data = await client.signAndSendContractOrScript(
+  const data = await signAndSendTransaction(
     txData.fromAddress,
     context.unsignedTxId,
-    context.unsignedTransaction.unsignedTx,
-    context.currentNetwork
+    context.unsignedTransaction.unsignedTx
   )
 
+  // TODO: Should we display a pending transaction when deploying a contract? What would the "toAddress" be in that case?
+  // Since there's no toAddress, we need to rethink our UI.
```

Comment:
In the next branch I am implenenting this by just showing a badge with the text `Smart contract` for the "to" address.